desktop/app/lib/keychain/index.js

```diff
@@ -1,26 +1,31 @@
-const keytar = require( 'keytar' );
+const { safeStorage } = require( 'electron' );
+const ElectronStore = require( 'electron-store' );
 
 const keychainService = 'WordPress.com';
+const store = new ElectronStore( {
+	name: keychainService,
+	watch: true,
+} );
 
 async function write( key, value ) {
-	return keytar.setPassword( keychainService, key, JSON.stringify( value ) );
+	if ( ! safeStorage.isEncryptionAvailable() ) {
+		throw new Error( 'Encryption is not avaialble.' );
+	}
+
+	const buffer = safeStorage.encryptString( value );
+	store.set( key, buffer.toString( 'latin1' ) );
 }
 
 async function read( key ) {
-	let value = await keytar.getPassword( keychainService, key );
-	if ( value ) {
-		value = JSON.parse( value );
+	if ( store.has( key ) ) {
+		const buffer = store.get( key );
+		return safeStorage.decryptString( Buffer.from( buffer, 'latin1' ) );
 	}
-	return value;
+	throw new Error( 'Requested value not found.' );
```

Comment:
I am guessing this is new behavior, as previously there was no check whether the target key exists in the keychain/store.